javatests/jsinterop/generator/externs/simpleclass/simpleclass.js

```diff
@@ -97,6 +110,15 @@ SimpleClass.prototype.thisType;
  */
 SimpleClass.prototype.fooMethod = function(foo, bar, opt_baz) {};
 
+/**
+ * @deprecated
+ * @param {string} bar
+ * @param {string} foo
+ * @param {Object=} opt_baz
+ * @return {boolean}
+ */
+SimpleClass.prototype.barMethod = function(bar, foo, opt_baz) {};
```

Comment:
in order to be consistent, I'd name this method `deprecatedMethod`

Could you also use one union type for parameter in order to test that JsOverlay methods are also marked as deprecated?

```js
/**
 * @deprecated
 * @param {string} bar
 * @param {string|number} foo
 * @param {Object=} opt_baz
 * @return {boolean}
 */
SimpleClass.prototype.deprecatedMethod = function(bar, foo, opt_baz) {};
```

Reply:
for this one i figured out that i need to add the annotation here https://github.com/google/jsinterop-generator/blob/746952f1741266362ca333db74537965e4fc88b3/java/jsinterop/generator/visitor/UnionTypeHelperTypeCreator.java#L175, or should i add it on the individual methods?
is there a nice way to get the annotation information of the original method to check if i should add the deprecation or not, or should i pass that in some new field/parameter.

Reply:
I think @jDramaix is only asking for a test that confirms that in the case of a union type param, that the various generated overlay methods are also correctly marked as deprecated.

Reply:
@niloc132 thank you, this means this one is also done.